public/app/features/query/components/QueryGroup.tsx

```diff
@@ -116,6 +119,9 @@ export class QueryGroup extends PureComponent<Props, State> {
           dataSource: { ...options.dataSource },
           savedQueryUid: options.savedQueryUid,
         },
+        // TODO: Detect the first panel added into a new dashboard better.
+        // This is flaky in case the UID is generated differently
+        isDataSourceModalOpen: locationService.getSearchObject().editPanel === '1',
```

Comment:
QueryGroup is designed to used outside dashboards, need to move this flow a bit. 

I would look into adding a new mode / state to panel edit, where we detect if this is the first panel, and then show this modal (and then we don't need to render anything but the modal)